packages/react-components/react-motion/library/src/factories/createMotionComponent.ts

```diff
@@ -94,18 +96,26 @@ export function createMotionComponent<MotionParams extends Record<string, Motion
 
       if (element) {
         const atoms = typeof value === 'function' ? value({ element, ...paramsRef.current }) : value;
-        onMotionStart();
 
         const handle = animateAtoms(element, atoms, { isReducedMotion: isReducedMotion() });
 
+        if (!disableMotions) {
+          onMotionStart();
+        }
+
+        if (disableMotions) {
+          handle.finish();
+          return;
+        }
+
         handle.setMotionEndCallbacks(onMotionFinish, onMotionCancel);
         handleRef.current = handle;
 
         return () => {
           handle.cancel();
         };
       }
-    }, [animateAtoms, handleRef, isReducedMotion, onMotionFinish, onMotionStart, onMotionCancel]);
+    }, [animateAtoms, handleRef, isReducedMotion, onMotionFinish, onMotionStart, onMotionCancel, disableMotions]);
```

Comment:
```suggestion
    }, [animateAtoms, handleRef, isReducedMotion, onMotionFinish, onMotionStart, onMotionCancel]);
```

Let's recreate `optionsRef` as in `createPresenceMotion` and store `disableMotions` there